database/src/main/scala/cromwell/database/SqlDatabase.scala

```diff
@@ -178,22 +178,12 @@ trait SqlDatabase extends AutoCloseable {
                                                   (implicit ec: ExecutionContext):
   Future[Traversable[(Execution, ExecutionInfo)]]
 
-  protected def addMetadataEvent(workflowUuid: String,
-                                 key: String,
-                                 value: String,
-                                 valueType: String,
-                                 timestamp: Timestamp)
-                                (implicit ec: ExecutionContext): Future[Unit]
-
-  protected def addMetadataEvent(workflowUuid: String,
-                                 key: String,
-                                 callFqn: String,
-                                 index: Option[Int],
-                                 attempt: Int,
-                                 value: String,
-                                 valueType: String,
-                                 timestamp: Timestamp)
-                                (implicit ec: ExecutionContext): Future[Unit]
+  /**
+    * Add metadata events to the database transactionally. normalized type structure is as follows:
+    * (WorkflowId, MetadataKey, Option[CallFqn, CallIndex, CallAttempt], MetadataValue, MetadataValueType, Timestamp)
+    */
+  protected def addMetadata(events: Iterable[(String, String, Option[(String, Option[Int], Int)], String, String, Timestamp)])
+                           (implicit ec: ExecutionContext): Future[Unit]
```

Comment:
@kshakir While I agree that overloaded `addMetadata` methods would be cleaner, in practice it's only going to result in more "useless" work to do since we'll have to partition the `MetadataEvent`s into "workflow events" and "call events", to choose the right `addMetadata` to call, which will both anyway call the same slick query in the end, only one with `Option(..)` and the other with `None` for some parameters. Or otherwise we could split Metadata into `WorkflowEvents` and `CallEvents` all the way up to the metadata service so the caller would have to choose what kind of event they're sending.